File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-31.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        int l = 0;
        HashSet<char> cSet = new HashSet<char>();
        int result = 0;
        for(int r = 0; r < s.Length; r++) {
            while(cSet.Contains(s[r])) {
                cSet.Remove(s[l]);
                l += 1;
            }
            cSet.Add(s[r]);
            result = Math.Max(result, r - l + 1);
        }
        return result;
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-31.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if (len(prices) <= 1):
            return 0
        res = 0
        l = 0
        r = 1
        while (r < len(prices)):
            res = max(prices[r] - prices[l], res)
            if (prices[l] >= prices[r]):
                l = r
            r+= 1
        return res

                